utils/check_repo.py

```diff
@@ -618,6 +619,20 @@ def find_tested_models(test_file: str) -> list[str]:
                     model_tested.append(name)
         return model_tested
 
+    # Models that inherit from `CausalLMModelTester` don't need to set `all_model_classes` -- it is built from other
+    # attributes by default.
+    if "CausalLMModelTester" in content:
+        all_models += re.findall(r"base_model_class\s+=.*", content)
+        all_models += re.findall(r"causal_lm_class\s+=.*", content)
+        all_models += re.findall(r"sequence_classification_class\s+=.*", content)
+        all_models += re.findall(r"question_answering_class\s+=.*", content)
+        all_models += re.findall(r"token_classification_class\s+=.*", content)
+        if len(all_models) > 0:
+            model_tested = []
+            for entry in all_models:
+                model_tested.append(entry.split("=")[1].strip())
+            return model_tested
```

Comment:
Our repo checks fail without these, the lines above expect `all_model_classes` to be explicitly set. But check the comment above the function, on L595-596: this requirement is due to a technical limitation, so it's okay to work around it.

This change prevents us from having to redefine `all_model_classes` on each model -- `CausalLMTester` already sets it 🤗